mongoengine/queryset/transform.py

```diff
@@ -98,6 +99,15 @@ def query(_doc_cls=None, **kwargs):
                         value = value['_id']
 
             elif op in ('in', 'nin', 'all', 'near') and not isinstance(value, dict):
+                if isinstance(value, BaseDocument):
```

Comment:
Both the checks mentioned in this [StackOverflow thread](http://stackoverflow.com/questions/19919314/how-to-check-isinstance-of-iterable-in-python) will say that BaseDocument is iterable. (`for k in user` will over each field of the document `user`)